quickstarts/09-quick-ref-agents/README.md

```diff
@@ -0,0 +1,278 @@
+# Quick Reference For Trying out Each Agent Type
+
+This quickstart demonstrates the unified agent experience with automatic agent type selection and YAML configuration support.
+It supports trying out the various agents and parameters supported.
+
+## Overview
+
+The unified agent interface provides a single entry point for all agent types with automatic selection based on configuration parameters:
+
+- **ToolCallAgent** (default): Simple, stateless tool execution
+- **ReActAgent**: Reasoning-action pattern with explicit thought processes
+- **OpenAPIReActAgent**: API integration with vector store capabilities
+- **DurableAgent**: Durable, stateful workflow with Dapr integration
+
+
+## Agent Type Selection Logic
+
+The system automatically selects the agent type based on configuration:
+
+1. **DurableAgent**: If `state_store_name` is specified
+2. **OpenAPIReActAgent**: If `openapi_spec_path` is specified
+3. **ReActAgent**: If `reasoning` is True
+4. **ToolCallAgent**: Default (simple tool execution)
+
+This provides a seamless experience where you don't need to understand the underlying agent types - the system chooses the best one automatically.
+
+## Prerequisites
+
+### Basic Requirements
+- **Python 3.10+** with pip
+- **OpenAI API Key** for LLM interactions
+- **Dapr Agents package** installed
+
+### For Durable Agents (DurableAgent)
+- **Dapr CLI** installed and initialized
+- Optionally, **Docker** (for Dapr components)
+
+## Setup
+
+### 1. Install Dapr Agents
+From the project root directory:
+```bash
+pip-compile pyproject.toml
+pip install -e .
+```
+
+### 2. Set Environment Variables
+Optionally,
+```bash
+export OPENAI_API_KEY="your-openai-key"
+```
+
+### 3. Install Dapr CLI (for durable agents only)
+
+Follow the instructions to install the latest [Dapr CLI](https://docs.dapr.io/getting-started/install-dapr-cli/). 
+
+## Running the Quickstarts
+
+Navigate to the quickstart directory:
+```bash
+cd quickstarts/09-quick-ref-all-agent-types
+```
+
+### Running Without Dapr (Stateless Agents)
+
+These examples work without Dapr and demonstrate the unified agent interface:
+
+#### 1. Simple Tool Agent
+```bash
+python 01_simple_tool_agent.py
+```
+**What it does**: Creates a basic weather assistant using ToolCallAgent (default)
+**Agent Type**: ToolCallAgent
+**Dapr Required**: No
+
+This is a simple agent that leverages specified `tools`, and is stateless.
+
+#### 2. Reasoning Agent
```

Comment:
I assume all of these exampls need to be updated too right? I am testing the `durable_agent.py` since the other ones do not exist anymore. Like the script for the `02_reasoning_agent.py`.